Merge from upstream v12

Alchez opened the pull request.

shariquerik committed:
fix: All todos not accessable to all users

(cherry picked from commit 1d628d0a13999c9c444fa9f5d2e2014e76d73237)

shariquerik committed:
fix: Also updated has_permission

(cherry picked from commit adf76a7ec931906bac8a7c6c6cf135ebb6aba79b)

shariquerik committed:
fix: added test case

(cherry picked from commit 0f1eb5c6b211855246749fc0c9ede2ed5d197d7b)

shariquerik committed:
fix: updated access logic and test case

(cherry picked from commit b7833ebb923685299bfdc75c1d7b94442c71bcb7)

shariquerik committed:
fix: Apply suggestions from code review

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
(cherry picked from commit e5c1682856e82cb5547a3781ad77c16da4d710ad)

shariquerik committed:
fix: updated get_doctype_roles logic based on suggestion

(cherry picked from commit 4caac4f90d501ee50704f43dbf8633ccd1bccc14)

surajshetty3416 committed:
fix: Typo

(cherry picked from commit c8ed2a1d660e20eae7086289ea87e41d48bc2cb7)

shariquerik committed:
fix: Apply suggestions from code review

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
(cherry picked from commit 08a58dc0c7ddac08cb39ea3626b0724a5a11736f)

shariquerik committed:
fix: updated test case

(cherry picked from commit 5e1a3a5d104d0177733e00118009ec7fe81217c7)

shariquerik committed:
fix: added test case to check read access to ToDo documents via role permissions

(cherry picked from commit a2e0a0cd7ea488fef81aca6584488d9d72051842)

shariquerik committed:
fix: updated test cases

(cherry picked from commit 3e61163c804337fa2de3e985710a2852cfd5190f)

shariquerik committed:
fix: minor fix

(cherry picked from commit b16e45040fe8ef4652eaf1febec82fece44bcc1a)

shariquerik committed:
fix: sider + updated test case

(cherry picked from commit 3d379990c0c6126f793be7c006752a3ca8cc640f)

shariquerik committed:
fix: sider + updated test case

(cherry picked from commit af810e8b6cc002072f4062a462d47038deab2608)

shariquerik committed:
refactor: relocated get_doctype_roles function

(cherry picked from commit 7aa062873194f4e4372e100aafbfd12a7eef4625)

shariquerik committed:
test: test case fix

(cherry picked from commit 700b91834784ea462581588567a4523157bc79d6)

navdeepghai committed:
feat(fixes): for frappeclient post_api(POST) request being redirected to GET request on calling server

(cherry picked from commit d73428df919dc98e7ee13af92af5fd8a66da66d4)

ankush committed:
fix: remove mutable defaults

mergify[bot] committed:
fix(minor): Don't allow changes to global hooks from server scripts (backport #14604) (#14606)

Co-authored-by: Aditya Hase <aditya@adityahase.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

gavindsouza committed:
Merge pull request #14613 from ankush/mut_def_v12

fix: remove mutable defaults

mergify[bot] committed:
fix: clean `join` param when executing `reportview.get` (backport #14845) (#14857)

* fix: clean `join` param when executing `reportview.get`

(cherry picked from commit 2e48c0a85ec99c68b640f4e9c7eea576409a8852)

# Conflicts:
#	frappe/desk/reportview.py

* fix: Resolve conflicts

Co-authored-by: Sagar Vora <sagar@resilient.tech>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

shadrak98 committed:
fix: add check to validate chat room members (#14848)

* fix: added check to validate members in chat room
* chore: Allow translations on validation message

Co-authored-by: gavin <gavin18d@gmail.com>

lukptr committed:
feat: prepared report read from replica

(cherry picked from commit e907f77ccd8369d42766925ab2ef5d3806d2e052)

# Conflicts:
#	frappe/desk/query_report.py

abdgad committed:
refactor: minor code cleanup in auto_email_report.py

(cherry picked from commit 29785af9158e3e28b435546b869af62df27869d2)

# Conflicts:
#	frappe/email/doctype/auto_email_report/auto_email_report.py

gavindsouza committed:
fix: Resolve conflicts

gavindsouza committed:
Merge pull request #14909 from frappe/mergify/bp/version-12-hotfix/pr-14883

feat: prepared report read from replica (backport #14883)

mergify[bot] committed:
fix: Multi level custom report fix (backport #14489) (#14873)

* fix: Multi level custom report fix

(cherry picked from commit 1e9056c0ee9994967d0bd6973dd2d3d6c8110370)

# Conflicts:
#	frappe/desk/query_report.py

* test: Added UI test for multi level query report

(cherry picked from commit bf0fb0de5c2bf83e293c48186a741917b6a7c031)

# Conflicts:
#	cypress/integration/query_report.js

* Resolved Conflicts in query_report.js UI test

* Resolved Conflicts in query_report.py

* fix: sider fix

Co-authored-by: Shariq Ansari <sharique.rik@gmail.com>
Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>

ci2014 committed:
Update interaction.js

Add assigned_by to interaction_values for ToDos to avoid PermissionError

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
style: Fix formatting issue

mergify[bot] committed:
fix: dont update `modified` for changes due to nsm implementation (backport #14928)

* fix: don't change `modified` when updating lft/rgt

These are implementation details and not explicitly modified by user.
IMO these `modified` value shouldn't change just because node has moved
around due to NSM change.

as

(cherry picked from commit fb929566ccee44599ed972ec6be64210b6fd423b)

# Conflicts:
#	frappe/utils/nestedset.py

* refactor(minor): better variable names

(cherry picked from commit 6a504429367024cddad336e1d5b05177b189f274)

# Conflicts:
#	frappe/utils/nestedset.py

* fix: resolve conflicts

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit 0c59b3b9905b854780f6dee71eb5453e1983b659)

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

ankush committed:
fix: unknown argument to set_value

surajshetty3416 committed:
Merge pull request #14907 from frappe/mergify/bp/version-12-hotfix/pr-14888

gavindsouza committed:
fix: Update Installed Applications on remove_app

Removing an app from a site should sync the Installed Applications
doctype

(cherry picked from commit 2e28ee5b25ad407d178b19b7efd1a2943603f073)

gavindsouza committed:
fix(cli): Add new line after each site's migrate

(cherry picked from commit ef39843f6ba8b24912fdb41a62df186dbb39bd15)

surajshetty3416 committed:
Merge branch 'version-12-hotfix' into patch-45

surajshetty3416 committed:
Merge pull request #14932 from ci2014/patch-45

surajshetty3416 committed:
Merge pull request #15029 from frappe/mergify/bp/version-12-hotfix/pr-15027

sagarvora committed:
fix: optimise `mark_email_as_seen`

sagarvora committed:
Merge pull request #15036 from resilient-tech/fix-seen-12

fix: optimise `mark_email_as_seen`

shariquerik committed:
fix: Remove chart from report when saved without chart

(cherry picked from commit 06aa8a10adb6dc05c91f2bc7dd589d1ea400b57d)

surajshetty3416 committed:
Merge pull request #15070 from frappe/mergify/bp/version-12-hotfix/pr-15064

gavindsouza committed:
fix: Don't pass kwargs that make_access_log can't handle

When people try to export reports, sometimes a cmd kwarg is passed
across the three wrapped functions right to make_access_log. This
happens because of the order and way we had to wrap the write_only,
retry and whitelist.

(cherry picked from commit ae9731467ac459b73d685d745f3bb2220384cf39)

# Conflicts:
#	frappe/core/doctype/access_log/access_log.py

surajshetty3416 committed:
Merge pull request #15086 from frappe/mergify/bp/version-12-hotfix/pr-15084

gavindsouza committed:
refactor: get_mapping_module doesn't need to access to instance

(cherry picked from commit 5a6d7ee191149291dc2b84391ca8fddeb9543388)

# Conflicts:
#	frappe/data_migration/doctype/data_migration_plan/data_migration_plan.py

gavindsouza committed:
Merge pull request #15097 from frappe/mergify/bp/version-12-hotfix/pr-15095

refactor: get_mapping_module doesn't need to access to instance (backport #15095)

nemesis189 committed:
feat: handle timeout and deadlocks in db.sql (#15092)

mergify[bot] committed:
fix: return self after submit/cancel (backport #15105) (#15108)

Co-authored-by: Ankush Menat <ankush@frappe.io>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

mergify[bot] committed:
fix: prevent report error for AttachmentLimitReached exceptions (backport #14515) (#15113)

Co-authored-by: Saqib Ansari <nextchamp.saqib@gmail.com>

mergify[bot] committed:
fix: Use separate API to insert route history (backport #15114) (#15117)

Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>

mergify[bot] committed:
refactor: Deferred insert for route history (backport #15120) (#15122)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

ankush committed:
fix: EPS enabled check causes querying full doc (#15144)

rohitwaghchaure committed:
chore: Merge branch 'version-12-hotfix' into 'v12-pre-release'

rohitwaghchaure committed:
Merge pull request #15145 from rohitwaghchaure/merge-hotfix-to-pre-release-for-12-24

chore: Merge branch 'version-12-hotfix' into 'v12-pre-release'

rohitwaghchaure committed:
Revert "fix: add check to validate chat room members (#14848)"

This reverts commit 2aee7bc62c84d285fef0b664ee7b8484a3862f3e.

(cherry picked from commit 9b62dec59bee7b07579041e39ddf8d8073642ebb)

rohitwaghchaure committed:
Revert "fix: add check to validate chat room members (#14848)"

This reverts commit 2aee7bc62c84d285fef0b664ee7b8484a3862f3e.

rohitwaghchaure committed:
Merge pull request #15167 from frappe/revert-14848-chat-room-fix

Revert "fix: add check to validate chat room members"

rohitwaghchaure committed:
Merge pull request #15168 from frappe/mergify/bp/v12-pre-release/pr-15167

Revert "fix: add check to validate chat room members" (backport #15167)

rohitwaghchaure committed:
chore: change log for v12.24

rohitwaghchaure committed:
Merge pull request #15169 from rohitwaghchaure/change-log-for-v12-24

chore: change log for v12.24

rohitwaghchaure committed:
Merge branch 'v12-pre-release' into version-12

rohitwaghchaure committed:
bumped to version 12.24.0

Commit:
Merge remote-tracking branch 'frappe/version-12' into merge-v12

ghost commented:
### Bug Fixes

* remove mutable defaults ([4751437](https://github.com/ParsimonyGit/frappe/commit/475143751c4f5024fb2db292a9b2b681610135c3))
*  **minor:** Don't allow changes to global hooks from server scripts (backport #14604) (#14606) ([cc8e4fb](https://github.com/ParsimonyGit/frappe/commit/cc8e4fba11a21c45a85edab086e506bc724d6921))
* clean `join` param when executing `reportview.get` (backport #14845) (#14857) ([7ea3a34](https://github.com/ParsimonyGit/frappe/commit/7ea3a343865dc82a7062f88a95ddc08e6d208000))
* add check to validate chat room members (#14848) ([2aee7bc](https://github.com/ParsimonyGit/frappe/commit/2aee7bc62c84d285fef0b664ee7b8484a3862f3e))
* Resolve conflicts ([5dec55b](https://github.com/ParsimonyGit/frappe/commit/5dec55b0695d5d8c66989a203defba2e0df51803))
* Multi level custom report fix (backport #14489) (#14873) ([e587e3a](https://github.com/ParsimonyGit/frappe/commit/e587e3a8c89dc59bf196e74e29ef7841b1014533))
* Resolve conflicts ([fb7dda0](https://github.com/ParsimonyGit/frappe/commit/fb7dda01335aae9325a68ede6f2fab82af342d0b))
* dont update `modified` for changes due to nsm implementation (backport #14928) ([3ef5bdf](https://github.com/ParsimonyGit/frappe/commit/3ef5bdf601b796dbba8a998a42a75b525438634a))
* unknown argument to set_value ([749f32b](https://github.com/ParsimonyGit/frappe/commit/749f32bb4e39d646dd34c4b8d60cc1f49d246635))
* Update Installed Applications on remove_app ([0689678](https://github.com/ParsimonyGit/frappe/commit/068967876db8acf2a12d41ed5bb1d1046f432859))
*  **cli:** Add new line after each site's migrate ([51fd418](https://github.com/ParsimonyGit/frappe/commit/51fd4183884403ca1b5d899a92619ab47643ae5f))
* optimise `mark_email_as_seen` ([b08e6d8](https://github.com/ParsimonyGit/frappe/commit/b08e6d81a48d52f05e0a8413f0ba67e6056eb8f0))
* Remove chart from report when saved without chart ([fbde416](https://github.com/ParsimonyGit/frappe/commit/fbde4169194088743fd39ba5a117b1a0c8372288))
* Don't pass kwargs that make_access_log can't handle ([7f0159f](https://github.com/ParsimonyGit/frappe/commit/7f0159f134e58f8e69c9317d402d42af4240c829))


### Code Refactoring

* minor code cleanup in auto_email_report.py ([48b1d1e](https://github.com/ParsimonyGit/frappe/commit/48b1d1ef45ddf9771a7a5ce4560991e7bcafd837))
* get_mapping_module doesn't need to access to instance ([ab7bbde](https://github.com/ParsimonyGit/frappe/commit/ab7bbde5feb3092b2a9ae1d3ac46caae8a323f1f))


### Features

* prepared report read from replica ([d4c10e2](https://github.com/ParsimonyGit/frappe/commit/d4c10e218b399100eb80662ab5dd7eee5afec868))
* handle timeout and deadlocks in db.sql (#15092) ([f5b0a1a](https://github.com/ParsimonyGit/frappe/commit/f5b0a1a83a3dc898cf4869ee9cd9fb0c04493058))


### Styles

* Fix formatting issue ([9bce65b](https://github.com/ParsimonyGit/frappe/commit/9bce65b1d7ac20ea43e34d297cb6a3760fecb7e5))


### Contributors

<a href="https://github.com/ankush">ankush</a>, <a href="https://github.com/mergify">mergify[bot]</a>, <a href="https://github.com/gavindsouza">gavindsouza</a>, <a href="https://github.com/shadrak98">shadrak98</a>, <a href="https://github.com/abdgad">abdgad</a>, <a href="https://github.com/lukptr">lukptr</a>, <a href="https://github.com/ci2014">ci2014</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/sagarvora">sagarvora</a>, <a href="https://github.com/shariquerik">shariquerik</a>, <a href="https://github.com/nemesis189">nemesis189</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

nextchamp-saqib committed:
fix: IndexError while handling sql timeout error

(cherry picked from commit 6f7d030e82ae3b2664f56a6fd9989cfbfc1f1648)

mergify[bot] committed:
fix: Clarified docstatus transition exceptions (#15194) (#15196)

* [fix] Clarified docstatus transition exceptions

Exceptions issued by the document.py `check_docstatus_transition` method are potentially very misleading. In cases where an invalid docstatus is used, users receive an confusing exception stating "Cannot change docstatus from 0 to 2" or "Cannot change docstatus from 1 to 0".

This PR adds an additional exception message when an invalid docstatus is used.

* fix: Clarified docstatus transition exceptions

Added additional clarifications to exception messages

(cherry picked from commit 8ead1d9c487b5f7f3e86016b0404a700cc11ba8c)

Co-authored-by: this-gavagai <this-gavagai@users.noreply.github.com>

surajshetty3416 committed:
Merge pull request #15222 from frappe/mergify/bp/version-12-hotfix/pr-12374

ankush committed:
Merge pull request #15365 from frappe/mergify/bp/version-12-hotfix/pr-15179

fix: IndexError while handling sql timeout error (backport #15179)

mergify[bot] committed:
fix(ui): Rendering forms created by disabled users (backport #15371) (#15390)

Co-authored-by: Gavin D'souza <gavin18d@gmail.com>

mergify[bot] committed:
fix: Rename Property Setter on Print Format Rename (backport #15458) (#15460)

Co-authored-by: Gavin D'souza <gavin18d@gmail.com>

surajshetty3416 committed:
fix: Remove type annotations to support python 2.7

the issue was introduced via https://github.com/frappe/frappe/pull/15458

mergify[bot] committed:
feat: option to not rebuild search on rename (backport #15443) (#15463)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

rohitwaghchaure committed:
chore: Merge branch 'version-12-hotfix' into 'v12-pre-release'

rohitwaghchaure committed:
Merge pull request #15479 from rohitwaghchaure/merge-hotfix-to-pre-release-for-12-25

chore: Merge branch 'version-12-hotfix' into 'v12-pre-release'

gavindsouza committed:
fix: Erroreneous query building when column exists in matching condition
and set field value

Given how we're still trying to refactor set_value for v14; instead of
diverging the fixes for each major version I'd rather change v12 to be
more similar to the next two versions instead.

So, I've pasted the changes from v14 as is. Repurcussions of this
include an extra select query on every set_value call...just as it is in
v13 and v14.

Once frappe.db.set_value is refactored, I plan to backport the perf
improvements to v13 and v12 so that the extra query isn't required.

(cherry picked from commit ed37d5299583096ade2d685f6ae1feefc0ca8408)

gavindsouza committed:
fix: Erroreneous query building when column exists in matching condition
and set field value

Given how we're still trying to refactor set_value for v14; instead of
diverging the fixes for each major version I'd rather change v12 to be
more similar to the next two versions instead.

So, I've pasted the changes from v14 as is. Repurcussions of this
include an extra select query on every set_value call...just as it is in
v13 and v14.

Once frappe.db.set_value is refactored, I plan to backport the perf
improvements to v13 and v12 so that the extra query isn't required.

rohitwaghchaure committed:
Merge pull request #15486 from gavindsouza/set_value-v12

fix: Erroreneous query building when column exists in matching condition and set field value

rohitwaghchaure committed:
Merge pull request #15489 from frappe/mergify/bp/v12-pre-release/pr-15486

fix: Erroreneous query building when column exists in matching condition and set field value (backport #15486)

mergify[bot] committed:
feat: Workflow transition condition allows datetime functions (backport #11899) (#15487)

Co-authored-by: conncampbell <conncampbell@conzu.ca>
Co-authored-by: Gavin D'souza <gavin18d@gmail.com>

pruthvi145 committed:
fix: clear cache when filters passed as dn param in `set_value`

(cherry picked from commit 289d7e7afa8c922d9069579d3a898ba98b529a9b)

rohitwaghchaure committed:
chore: change log for v12.25.0

rohitwaghchaure committed:
Merge pull request #15496 from rohitwaghchaure/change-log-for-v12-25

chore: change log for v12.25.0

rohitwaghchaure committed:
bumped to version 12.25.0

rohitwaghchaure committed:
Merge branch 'v12-pre-release' into version-12

Commit:
Merge remote-tracking branch 'frappe/version-12' into merge-v12

gavindsouza committed:
Merge pull request #15531 from frappe/mergify/bp/version-12-hotfix/pr-15492

fix: clear cache for all docs matched with the filters passed as `dn` param in `set_value` (backport #15492)

gavindsouza committed:
fix: Use params instead of data for client login

(cherry picked from commit 984420363a9aa8d60f95b197e0750e1766347ddf)

# Conflicts:
#	frappe/frappeclient.py

gavindsouza committed:
fix: Use format instead of f-strings, resolve conflicts

gavindsouza committed:
Merge pull request #15547 from frappe/mergify/bp/version-12-hotfix/pr-15542

fix: FrappeClient post_api request being redirected to GET request (backport #15542)

ahmadpak committed:
fix: email_string parsing when pulling email

(cherry picked from commit 6afa5c8da3c7d0e74425f572e39f042e76ae6d19)

# Conflicts:
#	frappe/core/doctype/communication/communication.py

ahmadpak committed:
refactor: communication.py

(cherry picked from commit 9ebfc1ced9cfb7643560858553e59243eb128ac4)

pruthvi145 committed:
fix: limit in `Document.get`

(cherry picked from commit d1852242a0acb37c799d00c2b9fc8ca6d71b3587)

# Conflicts:
#	frappe/model/base_document.py

pruthvi145 committed:
test: add test for limit

(cherry picked from commit a87899c2155a0c2576d551a31aa5f6535be6930b)

# Conflicts:
#	frappe/tests/test_document.py

gavindsouza committed:
Merge pull request #15576 from frappe/mergify/bp/version-12-hotfix/pr-15564

fix: communication method "get_contacts" (backport #15564)

gavindsouza committed:
Merge pull request #15584 from frappe/mergify/bp/version-12-hotfix/pr-15574

fix: limit in `Document.get` (backport #15574)

gavindsouza committed:
feat(minor): Add "in list view" fields in sort selector

(cherry picked from commit 7fa9a3fd39cbc5ecc4db8da50317c3927e302271)

gavindsouza committed:
refactor(minor): SortSelector.setup_from_doctype

(cherry picked from commit 5742217e8e680e52edc456f59870443398222e74)

surajshetty3416 committed:
Merge pull request #15713 from frappe/mergify/bp/version-12-hotfix/pr-15709

gavindsouza committed:
fix(Custom DocPerm): Use Link type instead of Data for parent (#15715)

* fix: Use Link type instead of Data for parent

This makes sure frappe.rename_doc updates linked doctype records

* Revert "fix: Use Link type instead of Data for parent"

This reverts commit 93b37351069df9a79f04bfa326378f09bf97700f.

* fix(rename_doc): Handle re-linking Custom DocPerm records

(cherry picked from commit d78b49620863dec0975212f0d3e57fdd1698a96c)

surajshetty3416 committed:
Merge pull request #15726 from frappe/mergify/bp/version-12-hotfix/pr-15715

hasnain2808 committed:
ci: type hints not supported in py2.7

hasnain2808 committed:
chore: remove type hints

rohitwaghchaure committed:
chore: merge branch 'version-12-hotfix' into 'v12-pre-release'

surajshetty3416 committed:
Merge pull request #15848 from frappe/hasnain2808-patch-1

surajshetty3416 committed:
Merge branch 'version-12-hotfix' of https://github.com/frappe/frappe into merge-branch-hotfix-to-pre-release-for-12-26

rohitwaghchaure committed:
Merge pull request #15849 from rohitwaghchaure/merge-branch-hotfix-to-pre-release-for-12-26

chore: merge branch 'version-12-hotfix' into 'v12-pre-release'

pruthvi145 committed:
fix: limit without filter

(cherry picked from commit 7ce9f1eaa1acd8c9d7ef618c06dc91182e8e5d93)
(cherry picked from commit 09f6d8e167b54cc91989e49e35df5dca24e10480)

gavindsouza committed:
Revert "feat: Workflow transition condition allows datetime functions (backport #11899) (#15487)"

This reverts commit bdf8a1f465126f7070cc0635a2c9c2529336c739.

(cherry picked from commit 486a2bcad87c4513613cc3b6d2b05708c852fa57)

rohitwaghchaure committed:
chore: change log for v12.26.0

rohitwaghchaure committed:
Merge pull request #15986 from frappe/mergify/bp/v12-pre-release/pr-15978

Revert "feat: Workflow transition condition allows datetime functions (backport #11899)" (backport #15978)

rohitwaghchaure committed:
Merge pull request #15987 from frappe/mergify/bp/v12-pre-release/pr-15984

fix: limit without filter (backport #15975) (backport #15984)

rohitwaghchaure committed:
Merge pull request #15988 from rohitwaghchaure/change-log-for-v12-26

chore: change log for v12.26.0

rohitwaghchaure committed:
Merge branch 'v12-pre-release' into version-12

rohitwaghchaure committed:
bumped to version 12.26.0

Commit:
Merge remote-tracking branch 'frappe/version-12' into merge-v12

ankush committed:
chore: pin MarkupSafe

ankush committed:
chore: bump version

Commit:
Merge remote-tracking branch 'frappe/version-12' into merge-v12
